### ISSUE TITLE
feathers generate service throws typeError: cannot read property 'jest' of undefined

### ISSUE THREAD

**Issue by franzherzog** [feathersjs/generator-feathers#644]
If you have no devDependency in your package.json the check in feathers-generator for jest will fail with an undefined object.

this line -> https://github.com/feathersjs/generator-feathers/blob/master/lib/generator.js#L40

temp fix:
npm install --save-dev jest

Error: 
(node:14423) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'jest' of undefined
    at ServiceGenerator.get testLibrary [as testLibrary] (/home/franzl/.nvm/versions/node/v14.18.0/lib/node_modules/@feathersjs/cli/node_modules/generator-feathers/lib/generator.js:40:59)

**Comment by franzherzog** [comment]
#https://github.com/feathersjs/cli/issues/176

**Comment by franzherzog** [comment]
#https://github.com/feathersjs/cli/issues/176

**Comment by franzherzog** [comment]
see also 
https://github.com/feathersjs/cli/issues/176
https://github.com/feathersjs/feathers/issues/2426

**Comment by daffl** [comment]
`devDependencies` and the chosen testing library are always added by the generator. If they are not added in a generated app it is usually because the initial yarn/npm installation failed for some reason. It should work again when generating a new app.

**Comment by franzherzog** [comment]
strange, seems there is problem with the  nodemon@^2.0.15 installation

but npm install nodemon@^2.0.15 in shell is working fine :thinking: 

this command generates a error message (see below)
```0 verbose cli [
0 verbose cli   '/home/franzl/.nvm/versions/node/v14.18.0/bin/node',
0 verbose cli   '/home/franzl/.nvm/versions/node/v14.18.0/bin/npm',  
0 verbose cli   'install',
0 verbose cli   'nodemon@^2.0.15',
0 verbose cli   'axios',
0 verbose cli   'eslint@^8.3.0',
0 verbose cli   'mocha@^9.1.3',
0 verbose cli   '--save-dev',
0 verbose cli   '--cache-min',
0 verbose cli   '86400'
0 verbose cli ]
```
```
verbose stack nodemon: No matching version found for nodemon@^2.0.15.
595 verbose stack     at module.exports (/home/franzl/.nvm/versions/node/v14.18.0/lib/node_modules/npm/node_modules/npm-pick-manifest/index.js:209:23)
```

**Comment by franzherzog** [comment]
feathers generate app output:
```? Do you want to use JavaScript or TypeScript? JavaScript
? Project name feathers-generator
? Description 
? What folder should the source files live in? src
? Which package manager are you using (has to be installed globally)? npm
? What type of API are you making? REST, Realtime via Socket.io
? Which testing framework do you prefer? Mocha + assert
? This app uses authentication No
? Which coding style do you want to use? ESLint
   create package.json
   create config/default.json
   create public/favicon.ico
   create public/index.html
   create .editorconfig
   create src/app.hooks.js
   create src/channels.js
   create src/index.js
   create src/logger.js
   create src/middleware/index.js
   create src/services/index.js
   create .gitignore
   create README.md
   create src/app.js
   create test/app.test.js
   create .eslintrc.json
   create config/production.json
   create config/test.json
npm WARN deprecated @types/socket.io-parser@3.0.0: This is a stub types definition. socket.io-parser provides its own type definitions, so you do not need this installed.
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)

added 166 packages, and audited 167 packages in 2s

8 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm ERR! code ETARGET
npm ERR! notarget No matching version found for nodemon@^2.0.15.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/franzl/.npm/_logs/2021-11-24T15_59_01_931Z-debug.log

> feathers-generator@0.0.0 lint
> eslint src/. test/. --config .eslintrc.json --fix "--fix"

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------```

**Comment by daffl** [comment]
You're right, I'm seeing an error as well but it's because of a mocha peer dependency. But that is basically the problem, if one of the installation steps fails all other dependencies don't get added and everything ends up in a weird state.

**Comment by daffl** [comment]
Should be fixed now in `@feathersjs/cli` v4.5.1

**Comment by franzherzog** [comment]
Thank you David.
5 years later and I am still hooked on feathers ... in a good way ;)
